### DISCUSSION TITLE
Set-PnPListItem: The specified user  could not be found.

### DISCUSSION THREAD

**Discussion by iFestral** [pnp/powershell#4009] [Q&A]
Hi,
For some reason, I cannot update a list item with multiple values with a variable.
I have columns with Person or Group types; for groups, it contains **Allow selection of Groups**. 
It works as expected manually but not with the PnP module.

Trying to use an array variable with groups where a user is a member:
```
$groups = "grp.user.dev.system_terraform|devops","grp.user.product_and_market|management"

Set-PnPListItem -List $sharepoint_list_name -Id $ListItemUpdate.ID -Values @{"Groups" = $groups}
```
Getting the error: Set-PnPListItem: The specified user  could not be found. 
PS tried to create New-PnPGroup for each.

When I'm using copy/paste data from the variable and put it into the value, everything works.
```
Set-PnPListItem -List $sharepoint_list_name -Id $ListItemUpdate.ID -Values @{"Groups" = "grp.user.dev.system_terraform|devops","grp.user.product_and_market|management"}
```

Any suggestions will be appreciated.

**Comment by AndersRask** [comment]
I think this is a known issue. It seems to work on lists but not on document libraries.

I added a Group field that allowed multiple values, and if i use Set-PnPListItem on a doclib it always only add the first item (no matter if i define an array in a variable or directly typing the values. If i do the exact same process on a List it works fine...

```
# both set only add first group
Set-PnPListItem -List "Documents" -Values @{"Groups" = "Everyone except external users","Test Members"} -Identity 1
$values = @("Everyone except external users","Test Members")
Set-PnPListItem -List "Documents" -Values @{"Groups" = $values} -Identity 1

# on a list it adds both
Set-PnPListItem -List "TestList" -Values @{"Groups" = $values} -Identity 1
```

**Reply by An-kun** [reply]
I have had this issue for 3-4 years, just dropped it last I had a bit of need for it.
Retried it today and the same issue appear, except its not for document library, it is with sharepoint lists.
I generally work around it with the pnpuser's id in a variable.  It works like 99/100 times, then you hit a specific item with a specific user and it just doesnt work. "The specified user could not be found."

Worth noting is that the issue only appears with variables. 
Set-PnPListItem -List "Documents" -Values @("Everyone except external users","Test Members")   <-- works fine

$values = @("Everyone except external users","Test Members")
Set-PnPListItem -List "Documents" -Values $values   <-- does not work

But like i mentioned, only with some list items and some users. Have experienced this in 2 different tenants before, but only have access to one now.

**Reply by FKatrin** [reply]
For me, the issue also appears when I directly pass the values, i.e. without variable, as in
`Set-PnPListItem -List "Documents" -Identity 1 -Values @{"Persons" = "mail1@contoso.com","mail2@contoso.com"}`

I tried so many methods, all to no avail. I'm going crazy about this issue. This is what I tested as well:

`Set-PnPListItem -List "Documents" -Identity 1 -Values @{"Persons" = "1","2"}`
`Set-PnPListItem -List "Documents" -Identity 1 -Values @{"Persons" = "1","2","1","2"}`
`Set-PnPListItem -List "Documents" -Identity 1 -Values @{"Persons" = 1,2}`
`Set-PnPListItem -List "Documents" -Identity 1 -Values @{"Persons" = $users}` --> where $users contains the user objects of both users via Get-PnPUser
`Set-PnPListItem -List "Documents" -Identity 1 -Values @{"Persons" = $usersJson}`  --> where $usersJson is the user data in a JSON format as retrieved via MS Graph /sites/[SITEID]/lists/[LISTID]/items/[ITEMID]

It's always the same result: only the first of both users is added. Why is this still a thing after all these years? There are multiple issues here about this problem, for example this one: [#1096](https://github.com/pnp/PnP-PowerShell/issues/1096) Some have existed since 2017. Can someone please fix this?

**Reply by FKatrin** [reply]
Stumbling across this issue again. Why is this topic being ignored? Can we expect at least a reply anytime soon?
